src/Jobs/GenerateCollectionSitemapJob.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Rapidez\Statamic\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Storage;
+use Rapidez\Statamic\Actions\GenerateSitemapAction;
+use Statamic\Eloquent\Entries\Entry;
+use Statamic\Sites\Site;
+use Statamic\Entries\Collection;
+
+class GenerateCollectionSitemapJob implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable, InteractsWithQueue, Queueable;
+
+    protected GenerateSitemapAction $sitemapGenerator;
+
+    public function __construct(
+        protected Site $site,
+        protected Collection $collection,
+    ) {
+        $this->sitemapGenerator = new (GenerateSitemapAction::class);
+    }
+
+    public function handle(): void
+    {
+        $sitemapContent = $this->sitemapGenerator->createSitemap($this->generateCollectionSitemap());
+
+        $sitemapPath = config('rapidez.statamic.sitemap.storage_directory')
+            . config('rapidez.statamic.sitemap.prefix')
+            . $this->site->attribute('magento_store_id')
+            . '_collection_'
+            . $this->collection->handle()
+            . '.xml';
+
+        Storage::disk('public')->put($sitemapPath, $sitemapContent);
+    }
+
+
+    protected function generateCollectionSitemap() : string
```

Comment:
Why not place this in the action?

Reply:
I'd rather keep the logic for getting the collection entries in the job, same goes for getting the terms in the term job. The action itself is just for building the sitemap based on the given urls.